src/controllers/eventEmitter/eventEmitter.ts

```diff
@@ -77,6 +83,54 @@ export default class EventEmitter {
     for (const cb of this.#errorCallbacks) cb(error)
   }
 
+  protected async withStatus(callName: string, fn: Function, allowMultipleActions = false) {
```

Comment:
consider renaming **allowMultipleActions** to **allowConcurrentActions** or **allowConcurrentActionCalls** and add some comments on what this prop is responsible for in its default state